### PR TITLE
Planner: chose the indexScan when the limit count cannot reach

### LINKED ISSUE #18460: Planner: choose the indexScan rather than TableScan when the limited count cannot reach
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
CREATE TABLE `audio_info` (   `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT 'id',    `status` int(11) NOT NULL DEFAULT '0' COMMENT 'status',   `url` varchar(255) NOT NULL DEFAULT '' COMMENT 'url',   PRIMARY KEY (`id`),    KEY `audio_info_status` (`status`) );

insert into audio_info values (1,1,'22'),(2,2,'1'),(3,3,'22'),(4,4,'1'),(5,5,'52'),(6,6, '62');

analyze table audio_info;


```
### 2. What did you expect to see? (Required)
use the index scan to avoid the full table scan.

### 3. What did you see instead (Required)
```
explain select * from audio_info where status = 4 limit 2;
+-------------------------------+---------+-----------+------------------+----------------------------------+
| id                            | estRows | task      | access object    | operator info                    |
+-------------------------------+---------+-----------+------------------+----------------------------------+
| Limit_8                       | 2.00    | root      |                  | offset:0, count:2                |
| └─TableReader_14              | 2.00    | root      |                  | data:Limit_13                    |
|   └─Limit_13                  | 2.00    | cop[tikv] |                  | offset:0, count:2                |
|     └─Selection_12            | 2.00    | cop[tikv] |                  | eq(test.audio_info.status, 4)    |
|       └─TableRangeScan_11     | 12.00   | cop[tikv] | table:audio_info | range:[0,+inf], keep order:false |
+-------------------------------+---------+-----------+------------------+----------------------------------+
5 rows in set (0.00 sec)
```
### 4. Affected version (Required)
mysql> select version();
+------------------------------------------+
| version()                                |
+------------------------------------------+
| 5.7.25-TiDB-v4.0.0-beta.2-737-g5f983b860 |
+------------------------------------------+
1 row in set (0.00 sec)

### 5. Root Cause Analysis
correcting the cost of full table scan is not right.

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close #18460 

Problem Summary: correct the row count estimation 

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test TODO
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 
- chose the indexScan when the limite count cannot reach

### CODE REVIEW

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -1227,9 +1227,15 @@ func (ds *DataSource) crossEstimateRowCount(path *util.AccessPath, expectedCnt f
 	if err != nil {
```
Comment (eurekaka) [inline planner/core/find_best_task.go:1227]: For your case, the `isFull` in line 1218 is expected to be true, so this function is supposed to return at line 1219?

Comment (fzhedu) [reply planner/core/find_best_task.go:1227]: no, it return false.

Comment (eurekaka) [reply planner/core/find_best_task.go:1227]: Why? it is not as expected, we should fix the problem there?

Comment (fzhedu) [reply planner/core/find_best_task.go:1227]: https://github.com/fzhedu/tidb/blob/5e0e508b4ba605231f1e6e17389dcbfef4ce7486/planner/core/find_best_task.go#L1165
ranges : [0, +oo)
rangeCounts[0] > 1
It seems a bug: https://github.com/fzhedu/tidb/blob/5e0e508b4ba605231f1e6e17389dcbfef4ce7486/planner/core/find_best_task.go#L1169 should before the `break` of #L1167

**Comment by fzhedu** [comment]
It is esstially caused by the estimation from CMsketch.

**Pull request closed without merging**